### PR TITLE
11239 option to disable slow dashboard things

### PR DESCRIPTION
**Pull request by smoyth**
Remaining hound issues are due to legacy code. I fixed a bunch of them.

Note also that some of the JS that you'll see in new files is copied from older ones. I didn't clean up all of it. So if you see things not matching our conventions or looking wonky that's probably why.

### CODE REVIEW

File: app/assets/stylesheets/local/_reports.scss [hunk 1]
```diff
@@ -1,44 +1,47 @@
...
+div.report-body table th { background-color: $super-light-gray;
   font-size: 90%;
   padding: 2px 5px 2px 2px; }
-div.report_body table td { background-color: $white; }
-div.report_body table th.col { text-align: $left;  }
+div.report-body table td { background-color: $white; }
+div.report-body table th.col { text-align: $left;  }
 
-div.report_body table th.row { text-align: $left;
+div.report-body table th.row { text-align: $left;
```
[the remark this answers is not in the document]
Comment (smoyth) [reply app/assets/stylesheets/local/_reports.scss:28]: This is legacy stuff I'm not going to fix ATM

File: app/assets/stylesheets/local/_reports.scss [hunk 2]
```diff
@@ -1,44 +1,47 @@
...
+div.report-body table td.number { text-align: right; }
 
-div.report_body table th { background-color: $super-light-gray;
+div.report-body table th { background-color: $super-light-gray;
   font-size: 90%;
   padding: 2px 5px 2px 2px; }
-div.report_body table td { background-color: $white; }
-div.report_body table th.col { text-align: $left;  }
+div.report-body table td { background-color: $white; }
+div.report-body table th.col { text-align: $left;  }
```
[the remark this answers is not in the document]
Comment (smoyth) [reply app/assets/stylesheets/local/_reports.scss:26]: This is legacy stuff I'm not going to fix ATM

File: app/assets/stylesheets/local/_reports.scss [hunk 3]
```diff
@@ -1,44 +1,47 @@
...
+div.report-body table td.string { text-align: left; }
+div.report-body table td.number { text-align: right; }
 
-div.report_body table th { background-color: $super-light-gray;
+div.report-body table th { background-color: $super-light-gray;
   font-size: 90%;
   padding: 2px 5px 2px 2px; }
-div.report_body table td { background-color: $white; }
-div.report_body table th.col { text-align: $left;  }
+div.report-body table td { background-color: $white; }
```
[the remark this answers is not in the document]
Comment (smoyth) [reply app/assets/stylesheets/local/_reports.scss:25]: This is legacy stuff I'm not going to fix ATM

File: app/assets/stylesheets/variables/_mixins.scss [hunk 4]
```diff
@@ -27,3 +27,16 @@
     vertical-align: middle;
   }
 }
+
+@mixin reviewed-response-col {
+  td.reviewed_col {
```
[the remark this answers is not in the document]
Comment (smoyth) [reply app/assets/stylesheets/variables/_mixins.scss:32]: This is legacy stuff I'm not going to fix ATM

File: app/assets/stylesheets/local/_reports.scss [hunk 5]
```diff
@@ -2,43 +2,54 @@
...
-div.report_body th.row_grouping_label > div { margin-left: 3px;
+div.report-body th.row_grouping_label > div { margin-left: 3px;
   overflow: hidden; }
-div.report_body th.col_grouping_label { text-align: center; }
+div.report-body th.col_grouping_label { text-align: center; }
 
 /* std form report */
-div.report_body {
-  table.report_meta {
+div.report-body {
```
[the remark this answers is not in the document]
Comment (smoyth) [reply app/assets/stylesheets/local/_reports.scss:43]: This is legacy stuff I'm not going to fix ATM

File: app/assets/stylesheets/local/dashboard/_recent_responses.scss [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  .recent-responses {
+    display: flex;
+    flex-direction: column;
+  }
+
+  .recent-responses-table {
+    flex: 1 0 0;
+    overflow-x: hidden;
+
+    table.index_table {
```
[the remark this answers is not in the document]
Comment (smoyth) [reply app/assets/stylesheets/local/dashboard/_recent_responses.scss:13]: This is legacy stuff I'm not going to fix ATM

File: app/assets/javascripts/legacy/views/dashboard.js [hunk 7]
```diff
@@ -22,17 +22,6 @@
       return false;
     });
 
-    // readjust stuff on window resize
-    $(window).on('resize', () => {
-      self.adjust_pane_sizes();
-      self.list_view.adjust_columns();
```
Comment (cooperka) [inline app/assets/javascripts/legacy/views/dashboard.js:28]: Yesss 🔥 🔥 🔥

File: app/assets/stylesheets/global/_layout.scss [hunk 8]
```diff
@@ -155,17 +162,16 @@ body {
 
 #content {
   background-color: $white;
-  min-height: 400px;
-  padding: 20px 25px 30px;
+  padding: 20px;
```
[the remark this answers is not in the document]
Comment (cooperka) [reply app/assets/stylesheets/global/_layout.scss]: Thoughts on disabling this hound? I don't personally find any value in alphabetizing CSS properties

File: app/assets/stylesheets/variables/_base.scss [hunk 9]
```diff
@@ -28,6 +28,7 @@ $even-lighter-gray: #f6f6f6;
 $light-background-color: $super-light-gray;
 $white: #fff;
 $gray-text: #777;
+$grayer-text: #ccc;
```
Comment (cooperka) [inline app/assets/stylesheets/variables/_base.scss:31]: To me "grayer" sounds darker than "gray" -- can we call it `$lighter-gray-text`?

File: app/views/dashboard/dashboard.html.erb [hunk 10]
```diff
@@ -12,8 +12,8 @@
   delete ELMO.app.dashboard;
 
   ELMO.app.dashboard = new ELMO.Views.DashboardView(<%=json(
-    # needed for storing the map bounds between refreshes
-    mission_id: current_mission.id,
+    mission_id: current_mission.id, # needed for storing the map bounds between refreshes
+    reloadInterval: Rails.env.development? ? 5 : 30, # shorter interval is better for debugging
```
Comment (cooperka) [inline app/views/dashboard/dashboard.html.erb]: This is great in general -- but the DOM gets reset every refresh which makes browser inspection more difficult. Is there an easy way to disable this in development? It's been a minor bother before but this will make it worse.

If no current method, my suggestion: support setting `window.noReload = true` via the browser console. I'm happy to submit a separate PR if you prefer.

**Review by cooperka — COMMENTED** [review]
Stellar improvements 🌟  A few small comments. Tested and verified in my browser too.

**Comment by cooperka** [comment]
Tangential to this PR, and unsure if we have much control over it: the map should zoom in when map is fullscreened otherwise it's fairly pointless to enlarge it. Can I file on redmine?

(this behavior was not modified by your changes here)

Screen Shot 2020-12-30 at 09 38 58

File: app/assets/stylesheets/global/_layout.scss (hunk 8 above)
[the remark this answers is not in the document]
Comment (smoyth) [reply app/assets/stylesheets/global/_layout.scss]: Yeah sure. I'll take care of it.

File: app/views/dashboard/dashboard.html.erb (hunk 10 above)
Comment (smoyth) [reply app/views/dashboard/dashboard.html.erb]: Yes perhaps you can do that when you next encounter it? I don't want to spend anymore time on this beast of an issue!

**Comment by smoyth** [comment]
Yes please do file the map thingy on Redmine and notify Mariam :)
